Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -996,6 +399,200 @@ bool CGImageIsMask(CGImageRef image) {
  @Notes
 */
 CGImageRef CGImageCreateWithMaskingColors(CGImageRef image, const CGFloat* components) {
+    RETURN_NULL_IF(!image);
     UNIMPLEMENTED();
     return StubReturn();
-}
\ No newline at end of file
+}
+
+#pragma region WIC_HELPERS
+
+CGImageRef _CGImageGetImageFromData(void* data, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_NULL, data, length);
+}
+
+CGImageRef _CGImageLoadGIF(void* bytes, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatGif, bytes, length);
+}
+
+CGImageRef _CGImageLoadBMP(void* bytes, size_t length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatBmp, bytes, length);
+}
+
+CGImageRef _CGImageLoadTIFF(void* bytes, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatTiff, bytes, length);
+}
+
+CGImageRef _CGImageLoadPNG(void* bytes, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatPng, bytes, length);
+}
+
+CGImageRef _CGImageLoadJPEG(void* bytes, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatJpeg, bytes, length);
+}
+
+CGImageRef _CGImageLoadImageWithWICDecoder(REFGUID decoderCls, void* bytes, int length) {
+    ComPtr<IWICImagingFactory> imageFactory = _GetWICFactory();
+
+    ComPtr<IWICBitmapDecoder> pDecoder;
+
+    if (decoderCls != GUID_NULL) {
+        ComPtr<IStream> spStream;
+        RETURN_NULL_IF_FAILED(::CreateStreamOnHGlobal(NULL, FALSE, &spStream));
+
+        ULONG written = 0;
+        HRESULT hr = spStream->Write(bytes, length, &written);
+        if (FAILED(hr) || (written != length)) {
+            TraceError(TAG, L"IStream::Write failed hr=%x len=%d written=%d", hr, length, written);
+            return NULL;
+        }
+
+        // seek to start
+        LARGE_INTEGER offset;
+        offset.QuadPart = 0;
+        RETURN_NULL_IF_FAILED(spStream->Seek(offset, STREAM_SEEK_SET, NULL));
+
+        RETURN_NULL_IF_FAILED(imageFactory->CreateDecoder(decoderCls, NULL, &pDecoder));
```

Comment:
hmm looks like refraction gone mad.

---
In reply to: [84762069](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r84762069) [](ancestors = 84762069,84755233)